sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DuplicateMetricStorageException.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.state;
+
+import io.opentelemetry.sdk.metrics.internal.descriptor.MetricDescriptor;
+
+/** There are multiple metrics defined with the same name/identity. */
+class DuplicateMetricStorageException extends IllegalArgumentException {
+
+  DuplicateMetricStorageException(
+      MetricDescriptor existing, MetricDescriptor next, String message) {
+    // TODO: Better error messages including async vs. sync instruments.
+    super(message + " Found: " + existing + ", Want: " + next);
+  }
+
+  private static final long serialVersionUID = 1547329629200005982L;
```

Comment:
this makes me extremely sad, that for some ungodly reason, we still need to put this cruft into our files.

Reply:
also, it needs to move to the top of the class.  ;)

Reply:
Agreed, fixed.